From 6f265b515e5b93cef9bef0c4994622574c55a1ba Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?P=C3=A1draig=20Brady?= <P@draigBrady.com>
Date: Fri, 7 Nov 2025 13:55:39 +0000
Subject: [PATCH] md5sum: fix --text with the MSYS2 runtime

Note the use of "rt" is non-standard, but we're restricting
its use here to systems that define O_BINARY, which should
invariably support "rt" mode.

* src/digest.c (): Where significant, explicitly use "rt" mode
with --text, as MSYS2 defaults to binary mode for fopen'd files
(though not for standard streams).
* NEWS: Mention the bug fix.
Fixes https://github.com/coreutils/coreutils/issues/123
---
 NEWS         | 6 ++++++
 src/digest.c | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/md5sum.c b/src/md5sum.c
index 58b9c9ec1..f330bc1e9 100644
--- a/src/md5sum.c
+++ b/src/md5sum.c
@@ -611,7 +611,7 @@ digest_file (char const *filename, int *binary, unsigned char *bin_result,
     }
   else
     {
-      fp = fopen (filename, (O_BINARY && *binary ? "rb" : "r"));
+      fp = fopen (filename, O_BINARY ? (*binary ? "rb" : "rt") : "r");
       if (fp == NULL)
         {
           if (ignore_missing && errno == ENOENT)
-- 
2.52.0

